substrate/frame/examples/basic/src/lib.rs

```diff
@@ -177,267 +188,199 @@ pub mod pallet {
+			balance: BalanceOf<T>,
+		},
+		/// The Dummy value has been set.
+		SetDummy {
+			/// The new value of Dummy.
+			balance: BalanceOf<T>,
+		},
+	}
+
+	// The call declaration block of our pallet. This states the entry points that we handle. The
 	// macro takes care of the marshalling of arguments and dispatch.
 	//
-	// Anyone can have these functions execute by signing and submitting
-	// an extrinsic. Ensure that calls into each of these execute in a time, memory and
-	// using storage space proportional to any costs paid for by the caller or otherwise the
-	// difficulty of forcing the call to happen.
-	//
-	// Generally you'll want to split these into three groups:
-	// - Public calls that are signed by an external account.
-	// - Root calls that are allowed to be made only by the governance system.
-	// - Unsigned calls that can be of two kinds:
-	//   * "Inherent extrinsics" that are opinions generally held by the block authors that build
-	//     child blocks.
-	//   * Unsigned Transactions that are of intrinsic recognizable utility to the network, and are
-	//     validated by the runtime.
+	// Each call must define a `#[pallet::weight(..)]` attribute to convey a set of
+	// static information about its dispatch. The FRAME System and FRAME Executive pallets then use
+	// this information to properly execute transaction, whilst keeping the total load of
+	// the chain in a moderate rate.
 	//
-	// Information about where this dispatch initiated from is provided as the first argument
-	// "origin". As such functions must always look like:
+	// The parenthesized value of the `#[pallet::weight(..)]` attribute can be any type that
+	// implements the following traits:
+	// - [`WeighData`]: conveys the weight (a numeric representation of pure
+	// execution time and difficulty) of the transaction
+	// - [`ClassifyDispatch`]: demonstrates the [`DispatchClass`] of the call
+	// - [`PaysFee`]: indicates whether an extrinsic must pay transaction fees or not
 	//
-	// `fn foo(origin: OriginFor<T>, bar: Bar, baz: Baz) -> DispatchResultWithPostInfo { ... }`
+	// Larger weights imply larger execution time a block can handle (less of which can be placed in
+	// a single block).
 	//
-	// The `DispatchResultWithPostInfo` is required as part of the syntax (and can be found at
-	// `pallet_prelude::DispatchResultWithPostInfo`).
-	//
-	// There are three entries in the `frame_system::Origin` enum that correspond
-	// to the above bullets: `::Signed(AccountId)`, `::Root` and `::None`. You should always match
-	// against them as the first thing you do in your function. There are three convenience calls
-	// in system that do the matching for you and return a convenient result: `ensure_signed`,
-	// `ensure_root` and `ensure_none`.
+	// The weight for these calls relies on `WeightInfo`, which is auto-generated from the
+	// benchmark toolchain.
 	#[pallet::call(weight(<T as Config>::WeightInfo))]
 	impl<T: Config> Pallet<T> {
```

Comment:
reviewed up until this point. Will do another pass once existing comments are addressed.